### PR TITLE
auth: remove attodot feature; fixes #10254

### LINKED ISSUE #10254: auth: attodot breaks presigned zones with @ in SOA
A user had a zone containing something like:

```
7bits.nl.		300	IN	SOA	lorentz.7bits.nl. hostmaster\@7bits.nl. 129 600 3600 604800 3600
```

plus an RRSIG over that. They loaded this zone into PowerDNS, as presigned. PowerDNS served it, and helpfully converted `\@` to `.`, making the RRSIG invalid.

We should either drop the `attodot` feature or make it smarter.

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Fixes #10254.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/dnsbackend.cc [hunk 1]
```diff
@@ -340,7 +340,7 @@ void fillSOAData(const string &content, SOAData &data)
 
   try {
     data.nameserver = DNSName(parts.at(0));
-    data.hostmaster = DNSName(attodot(parts.at(1))); // ahu@ds9a.nl -> ahu.ds9a.nl, piet.puk@ds9a.nl -> piet\.puk.ds9a.nl
+    data.hostmaster = DNSName(parts.at(1)); // ahu@ds9a.nl -> ahu.ds9a.nl, piet.puk@ds9a.nl -> piet\.puk.ds9a.nl
```
Comment (rgacogne) [inline pdns/dnsbackend.cc:343]: The comment should likely go as well :-)

Comment (Habbie) [reply pdns/dnsbackend.cc]: yes!

**Comment by Habbie** [comment]
`check-zone` should notice `@` too.

**Comment by Habbie** [comment]
> `check-zone` should notice `@` too.

done

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -370,6 +370,12 @@ static int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, con
         cout<<"[Warning] SOA autocomplete is deprecated, missing field(s) in SOA content: "<<rr.qname<<" IN " <<rr.qtype.toString()<< " '" << rr.content<<"'"<<endl;
       }
 
+      if(parts.size() >= 2) {
+        if(parts[1].find('@') != string::npos) {
+          cout<<"[Warning] Found @-sign in SOA RNAME, should probably be a dot (.): "<<rr.qname<<" IN " <<rr.qtype.toString()<< " '" << rr.content<<"'"<<endl;
```
Comment (mind04) [inline pdns/pdnsutil.cc:375]:
```suggestion
          cout<<"[Warning] Found @-sign in SOA RNAME, should probably be a dot (.): "<<rr.qname<<" IN " <<rr.qtype.toString()<< " '" << rr.content<<"'"<<endl;
          numwarnings++;
```

Comment (Habbie) [reply pdns/pdnsutil.cc:375]: Somehow you managed to suggest a change that changes nothing. Impressive. Looks like a GitHub bug :-)

Comment (Habbie) [reply pdns/pdnsutil.cc:375]: oh, you added a line.

**Pull request merged**